artiq/applets/progress_bar.py

```diff
@@ -0,0 +1,38 @@
+#!/usr/bin/env python3
+
+from PyQt5 import QtWidgets
+
+from artiq.applets.simple import SimpleApplet
+
+
+class ProgressWidget(QtWidgets.QProgressBar):
+    def __init__(self, args):
+        QtWidgets.QProgressBar.__init__(self)
+        self.setMaximum(args.max)
```

Comment:
I don't quite get this. Is the maximum set by a dataset or the command line argument? Why is handling the maximum different than handling the minimum?

Reply:
Imagine using it to track the number of `ScanPoint` instances left in the run. This will possibly change every run and can thus be broadcasted as a dataset. In case your max is not set dynamically, the user should be able to set fix min/max values. I agree, behaves strange in case the user does not provide a second dataset. Will have a look at this particular case.

Reply:
Resolved in 7141975d5e06c9c7fb095e355e6081fdd24eaff9
tl:dr I removed the mechanic. Now there is only one dataset - the value - the rest is set once on startup